### PR TITLE
Add documentation on how to do incremental builds

### PR DESCRIPTION
**Pull request by pcmoritz**
On my system, this reduces compilation time from 2:40 min to 0:30 min, therefore enabling faster development iterations.

### CODE REVIEW

**Comment by pcmoritz** [comment]
One thing I noticed is the old `python setup.py develop` is also doing fast incremental compilation, but unfortunately that's deprecated and probably most people don't use it any more (e.g. it prints a big warning banner that it shouldn't be used any more and links to https://github.com/pypa/setuptools/issues/917).

The reason why `python setup.py develop` is fast is that it uses a stable temporary directory for the build, and it also uses the system `torch` path by default -- basically what this PR hacks `pip install -e .` to to also :)

**Review by zhuohan123 — APPROVED** [review]
Thanks for the feature! Left a small comment

File: setup.py [hunk 1]
```diff
@@ -22,6 +22,9 @@
 ROCM_SUPPORTED_ARCHS = {"gfx90a", "gfx942"}
 # SUPPORTED_ARCHS = NVIDIA_SUPPORTED_ARCHS.union(ROCM_SUPPORTED_ARCHS)
 
+if "VLLM_INCREMENTAL_BUILD_TORCH_PATH" in os.environ:
```
[the remark this answers is not in the document]
Comment (pcmoritz) [reply setup.py:31]: Thanks, I added the docs, let me know what you think :)

[the remark this answers is not in the document]
Comment (pcmoritz) [reply setup.py:31]: I also tried to do `VLLM_INCREMENTAL_BUILD=1` to make it easier btw, but I couldn't figure out how to get the global torch path from within `setup.py` -- I tried both via python utilities and using `subprocess` (both with and without shell), but I think there is some environment variable set which points the package dir to the temporary directory. It is probably best not to try to hack around this further :)

**Comment by pcmoritz** [comment]
I was thinking about this a little more, maybe it is best to just document ’python setup.py develop' since it has even less overhead. The other approach uses an internal variable in torch so not clear it is better. I'll make that change and ping you!

**Review by zhuohan123 — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**